test/uk/gov/hmrc/transitmovements/controllers/ArrivalsControllerSpec.scala

```diff
@@ -360,4 +362,41 @@ class ArrivalsControllerSpec
     }
   }
 
+  "getArrivalMessages" - {
+
+    val request = FakeRequest("GET", routes.ArrivalsController.getArrivalMessages(eoriNumber, movementId).url)
+
+    "must return OK and a list of message ids" in {
+      val messageResponses = MessageResponse.fromMessageWithoutBody(movement.messages.head)
+
+      lazy val messageResponseList = NonEmptyList.one(messageResponses)
+
+      when(mockRepository.getMessages(EORINumber(any()), MovementId(any()), eqTo(MovementType.Arrival), eqTo(None)))
```

Comment:
We need to ensure that if the `receivedSince` parameter is passed down that it continues to be passed down to the repository.

We could use property based testing for this (using ScalaCheck and forall like you've hopefully seen in other tests - we should randomly generate the `OffsetDateTime` using a generator and use 

```
"must return OK and a list of message ids" in forall(Gen.option(dateTimeGenerator)) { 
    dateTimeMaybe => 
       ...
       when(mockRepository.getMessages(EORINumber(any()), MovementId(any()), eqTo(MovementType.Arrival), eqTo(dateTimeMaybe)))
      ...
```

Equally, the request should be generated for each test in this case, rather than global.

Reply:
Done the changes as suggested above.